lib/utils.ts

```diff
@@ -1,12 +1,16 @@
 import { promisify } from "es6-promisify";
 import abi = require("ethereumjs-abi");
 import TruffleContract = require("truffle-contract");
-import Web3 = require("web3");
+import { providers as Web3Providers, Web3 } from "web3";
 import { gasLimitsConfig } from "../gasLimits.js";
 import { Address, DefaultSchemePermissions, Hash, SchemePermissions } from "./commonTypes";
 import { ConfigService } from "./configService";
 import { TransactionReceiptTruffle } from "./contractWrapperBase";
 import { LoggingService } from "./loggingService";
+// haven't figured out how to get web3 typings to properly expose the Web3 constructor.
```

Comment:
you could just do 
```
declare namespace NodeJS {
  export interface Global {
    web3: any
  }
}
```

Reply:
The problem isn't with accessing `web3`, it is with accessing `Web3` as a constructor (note the capital 'W').  I don't want to define web3 globally anyways, as web3 is not always available to this particular code, and I don't want to risk errors occurring from any code assuming that web3 is necessarily available.

The problem here is a compiler code-generation issue that manifests at runtime -- neither the compiler nor intellisense complain, but the compiler generates the wrong code for the constructor function.  Web3 sucks with respect to how it is structured as a module and in any way that facilitates knowing the typings.  Don't suggest web3-typescript-typings because it is worthless.

For this PR I created my own web3.d.ts and they are very helpful (one for lib and one for test code).  There is a lot of redundancy between those two d.ts files, but I expect this is temporary as I am hoping that web3 1.0 will have much better support for typings out-of-the-box.